keystore/src/entities/platform/generic/proteus/identity.rs

```diff
@@ -110,17 +111,15 @@ impl EntityBase for ProteusIdentity {
         debug_assert!(count <= 1);
         Ok(count)
     }
+}
 
-    async fn save(&self, conn: &mut Self::ConnectionType) -> crate::CryptoKeystoreResult<()> {
-        if Self::count(conn).await? == 1 {
-            return Ok(());
-        }
-
-        let transaction = conn.transaction()?;
-
+#[cfg_attr(target_family = "wasm", async_trait::async_trait(?Send))]
+#[cfg_attr(not(target_family = "wasm"), async_trait::async_trait)]
+impl EntityTransactionExt for ProteusIdentity {
+    async fn save(&self, transaction: &TransactionWrapper<'_>) -> crate::CryptoKeystoreResult<()> {
         use rusqlite::ToSql as _;
         transaction.execute(
-            "INSERT INTO proteus_identities (sk, pk) VALUES (?, ?)",
+            "INSERT INTO proteus_identities (sk, pk) SELECT ?, ? WHERE (SELECT COUNT(*) FROM proteus_identities) = 0",
```

Comment:
What is the reason for this change?

Reply:
To keep the logic that was present on current main, which would do an early return in this function:
```
if Self::count(conn).await? == 1 {
            return Ok(());
        }
```

Reply:
Just realized that this won't work as expected if there is a record in the persisted DB but none in the cache.

Reply:
Not an issue because the calling site takes care of the case that is handled here. Resolving.